### PR TITLE
offset the anchor scrolls by the fixed header size

### LINKED ISSUE #648: Clicking link in left nav scrolls doc too far up
Clink link in left nav.

Expected: content is displayed visibly and below the top nav.

Actual: content is displayed underneath the top nav.

### PR DESCRIPTION
**Pull request by devoncarew**
Fix #648 - offset the anchor scrolls by the fixed header size.

### CODE REVIEW

**Comment by devoncarew** [comment]
@sethladd - tried the css solution, but it interfered with our rendering. JS it is -

**Comment by sethladd** [comment]
web programming is awesome!!!

lgtm

File: lib/templates/_footer.html [hunk 1]
```diff
@@ -47,10 +47,16 @@
   }
 }
 
+function shiftWindow() {
```
Comment (sethladd) [inline lib/templates/_footer.html]: maybe leave a comment about what this is fixing and why this is here. Link to the bug?

File: lib/templates/_footer.html [hunk 2]
```diff
@@ -47,10 +47,16 @@
...
+}
+
 document.addEventListener("DOMContentLoaded", function() {
-    prettyPrint();
-    initScroller();
-    initSideNav();
+  prettyPrint();
+  initScroller();
+  initSideNav();
+  if (location.hash) shiftWindow();
```
Comment (sethladd) [inline lib/templates/_footer.html]: comment here, too, about why this exists. :)

**Comment by sethladd** [comment]
one issue with this solution... (This isn't a huge problem) but for those anchors near the bottom of the page, no shifting/scrolling offset is needed. However, even the anchors at the bottom of the page are shifted, so they are shifted down farther than they need to.

This is visible in the core SDK docs, e.g., with the SDK/package listing a lot of libraries.

**Pull request merged**